### ISSUE TITLE
Wrong owner for schema

### ISSUE THREAD

**Issue by hahnn** [CrunchyData/postgres-operator#1334]
**Our context:**
We've created test databases using the Crunchy operator, but not using the pgo client: we did that using CRDs directly and Vanilla kubernetes (with the help provided by your blog post https://info.crunchydata.com/blog/creating-a-postgresql-cluster-with-kubernetes-crds).

**Bug description:**
When a new database is created, the operator automatically creates an associated schema inside the database having the name of the user.
But, the owner of this schema is the one of the database superuser (postgres) which prevent the "standard" user to use it. We have to change the owner of this schema with the ALTER SCHEMA sql command.
But maybe it's by design?

**Expected behavior:**
When a new database is created by the operator, with its users, the schema created inside this database should be owned by the standard user and not by the postgres user.
(And BTW, if the operator could allow the name of the schema to be defined via a configuration variable, that would be nice.)

**Environment:**
Operator Version: 4.2.2

* Operating System: CentOS 7
* Where is this running: Cloud Provider (Hidora)
* Storage being used:  NFS
* Container Image Tag: centos7-12.2-4.2.2
* PostgreSQL Version: 12.2
* Platform: Kubernetes
* Platform Version: 1.15.5

**Comment by jkatz** [comment]
Hi @hahnn,

Yup, that sounds like it would defeat the purpose of having the user schema. I have patched `master` and backpatched to the 4.2 line that now gives the regular user authorization to use their own schema, create objects in it, etc.

In terms of customizing / adding more schemas, I think this gets into a much longer discussion. First, the intended behavior of what was supposed to be the feature was to match general PostgreSQL guidance around the [search path](https://www.postgresql.org/docs/current/ddl-schemas.html#DDL-SCHEMAS-PATH) and in light of the [issues surrounding the public schema relative to search path](https://info.crunchydata.com/blog/demystifying-schemas-search_path-through-examples), it is a best practice to have the user schema define.

Now, we could give the standard user "CREATE" rights within the database, and then the user can add/remove their own schemas. I happen to be looking at this code right now for a separate set of features, so I will consider that ,ensuring there is a proper balance between "getting a PostgreSQL database up and running" with "hardening for production deployments"

The good news is the principal issue is now resolved and the fix will appear in 4.3 and 4.2.3.

Thanks for reporting!
